GoogleForCreators/web-stories-wp
# Dashboard: Date formatting util to handle relative date display

## Summary
Pulling out the `getFormattedDisplayDate` util from Dashboard/utils and creating a local "package" to eventually get consumed by both editor and dashboard to remove dependency on `@wordpress/date`. This came about because the dashboard is parsing dates on stories and timezones are getting mixed up. This PR paves the way to resolve that bug. 

## Relevant Technical Choices
- Pulled apart all the various functionality in `getFormattedDisplayDate` to make testing more robust and make functions easier to reuse. 
- Use .parseZone in Moment formatting to honor the existing timezones passed in and not touch any browser specific timezone manipulation
- custom getTimeFromNow instead of using moment's .fromNow() - .fromNow uses the browser's timezone! We want to avoid this since users can set whatever timezone they want!
- getTimeFromNow will use getDateObjectWithTimezone as a source of true time for the date we're finding time from or until. This function makes a new utc date and then uses moment-timezone to convert that to the setting specific timezone and then we format it so that that date/time is what is seen and used to compare.

## To-do
- Separate implementation of this in dashboard to avoid a 40 file PR. :D  

## User-facing changes
- None 

## Testing Instructions
- Ensure that new date/tests are running and make sense to you.

---



Addresses #2199
